Tasks/DotNetCoreInstallerV1/Task Design.md

```diff
@@ -0,0 +1,110 @@
+- User can build all applications in a project using a single build task and need not create different pipelines or add multiple build task.
+
+### How it works:
+This feature is implemented, by asking user a new input: `installationPath`, the path where .Net core SDK/runtime should be installed. The Agent process should have write access on the path.
+There are multiple benefits of asking `installationPath` from user:
+- User can choose to install a particular version in isolation of all other version already installed, by specifying a path which is cleaned up after each release/build.
+- User can augment existing set of versions installed with another version by specifying a path where a set of .Net core versions might already be installed, such as: $(Agent.ToolsDirectory)
+- User can also install it at global .Net Core installation path, such as: C:\Program Files\dotnet or /usr/share/dotnet/. This way all agents running on that machine will have access to the newly added .Net core versions.
+
+The way a version is installed at the `installationPath` is now as follows:
+- We extract all folders from dotnetVersions.zip/tgz into the `installationPath`, just like install-dotnet.ps1/sh.
+- The files in root directory of archive are only copied/overriden if
+  - The .Net core version being installed at `installationPath` is later than all other versions already installed at the path.
+
+The advantage of the above installation approach are:
+- Time is saved by not copying unneeded files.
+- Failures which might occur while overriding existing files are avoided.
+
+Also to furhter optimize and save time while installation, we do not install a version if it is already cached at that path. Below described are different approaches to find out if a version is cached,
+
+1. **`Folder based approach`**
+   - `Installation process:` While installing a version, we will only extract the files from downloaded archive and then copy paste the files into the installationPath folder.
+   - `Caching process:` Existance of folder with name = version inside sdk or host/fxr folder will be considered as proof that sdk/runtime version is cached. Alogirthm to find if version is cached or not will be like this:
+     - `installationPath` should contains a folder called `sdk or host/fxr`. If it
+       - `exists:` check if sdk or host/fxr folder contains a folder with name = version to be downloaded.
+         - `exists:` Version is cached, nothing more needs to be done.
+         - `doesn't exist:` Version needs to be installed.
+       - `doesn't exist:` Version needs to be installed
+
+2. **`version.completed file based approach`**
+   - `Installation process:` Once all the files are successfully extracted, copied and pasted into installationPath, we will create a **`version.completed`** file in sdk/runtime folder. The version.completed file will exist at the same level as the version folder and will be considered as a stamp of successfull installation.
```

Comment:
why in sdk/runtime folder? why not in installationPath directly? The marker file can be named version.[sdk|runtime].completed

Reply:
It seemed more organised and clean that way